src/apps/sommelier/positions.ts

```diff
@@ -0,0 +1,101 @@
+import { logger } from '../../log'
+import { getClient } from '../../runtime/client'
+import { getTokenId } from '../../runtime/getTokenId'
+import { toDecimalNumber } from '../../types/numbers'
+import {
+  PositionDefinition,
+  PositionsHook,
+  PricePerShareContext,
+  UnknownAppTokenError,
+} from '../../types/positions'
+import { cellarV0821Abi } from './abis/cellar'
+import { getSommStrategiesData } from './sommelierApi'
+
+const hook: PositionsHook = {
+  getInfo() {
+    return {
+      id: 'sommelier',
+      name: 'Sommelier',
+      description: 'Sommelier vaults',
+    }
+  },
+  async getPositionDefinitions({ networkId }) {
+    const client = getClient(networkId)
+    const cellars = await getSommStrategiesData(networkId)
+    const positionDefinitions: PositionDefinition[] = []
+
+    const results = await Promise.allSettled(
+      cellars.map(async (cellar) => {
+        const [underlyingAsset, underlyingAssetSymbol, cellarName] =
+          await Promise.all([
+            client.readContract({
+              address: cellar.address,
+              abi: cellarV0821Abi,
+              functionName: 'asset',
+            }),
+            client.readContract({
+              address: cellar.address,
+              abi: cellarV0821Abi,
+              functionName: 'symbol',
+            }),
+            client.readContract({
+              address: cellar.address,
+              abi: cellarV0821Abi,
+              functionName: 'name',
+            }),
+          ])
+
+        return {
+          type: 'app-token-definition' as const,
+          pricePerShare: async ({ tokensByTokenId }: PricePerShareContext) => {
+            const tokenId = getTokenId({
+              address: cellar.address,
+              networkId,
+            })
+            const { decimals } = tokensByTokenId[tokenId]
+            return [toDecimalNumber(BigInt(cellar.shareValue), decimals)]
+          },
+          networkId,
+          address: cellar.address,
+          tokens: [
+            {
+              address: underlyingAsset.toLowerCase(),
+              networkId,
+            },
+          ],
+          displayProps: () => {
+            return {
+              title: cellarName,
+              description: underlyingAssetSymbol,
```

Comment:
In a future improvement, it could be nice to include the APY in the description. 💡

Reply:
ahah yes I imagine it wasn't directly obvious.

Reply:
ah, i think i found the logic that they use to derive APY. i will add this as a follow up